src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -544,9 +566,58 @@ zocl_load_sect(struct drm_zocl_dev *zdev, struct axlf *axlf,
 	case BITSTREAM_PARTIAL_PDI:
 		ret = zocl_load_partial(zdev, section_buffer, size);
 		break;
+	case PARTITION_METADATA:
+		if (zdev->partial_overlay_id != -1 && axlf->m_header.m_mode == XCLBIN_PR) {
+			err = of_overlay_remove(&zdev->partial_overlay_id);
+			if (err < 0) {
+				DRM_WARN("Failed to delete rm overlay (err=%d)\n", err);
+				ret = err;
+				goto OUT;
+			}
+			zdev->partial_overlay_id = -1;
+		} else if (zdev->full_overlay_id != -1 && axlf->m_header.m_mode == XCLBIN_FLAT) {
+			err = of_overlay_remove_all();
+			if (err < 0) {
+				DRM_WARN("Failed to delete static overlay (err=%d)\n", err);
+				ret = err;
+				goto OUT;
+			}
+			zdev->partial_overlay_id = -1;
+			zdev->full_overlay_id = -1;
+		}
+
+		bsize = zocl_read_sect(BITSTREAM, &bsection_buffer, axlf, xclbin);
+		bo = zocl_drm_create_bo(zdev->ddev, bsize, ZOCL_BO_FLAGS_CMA);
+		if (IS_ERR(bo)) {
+			vfree(bsection_buffer);
+			ret = -1;
+			goto OUT;
+		}
+		vaddr = bo->cma_base.vaddr;
+		memcpy(vaddr,bsection_buffer,bsize);
+
+		zdev->fpga_mgr->dmabuf = drm_gem_prime_export(&bo->gem_base, 0);
+		err = of_overlay_fdt_apply((void *)section_buffer, size, &id);
+		if (err < 0) {
+			DRM_WARN("Failed to create overlay (err=%d)\n", err);
+			zdev->fpga_mgr->dmabuf = NULL;
+			zocl_drm_free_bo(bo);
+			vfree(bsection_buffer);
+			ret = err;
```

Comment:
is a return missing? Or you may want to goto out  after freeing bo and setting the ret?

Reply:
fixed